Content.Shared/_RMC14/Medical/Pain/PainSystem.cs

```diff
@@ -0,0 +1,189 @@
+    }
+
+    public void AddPainModificator(EntityUid uid, PainModificator mod, PainComponent? pain = null)
+    {
+        if (!Resolve(uid, ref pain))
+            return;
+
+        pain.PainModificators.Add(mod);
+        UpdateCurrentPainPercentage(pain);
+        Dirty(uid, pain);
+
+        Timer.Spawn(mod.Duration, () => RemovePainModificator(uid, mod, pain));
+    }
+
+    private void RemovePainModificator(EntityUid uid, PainModificator mod, PainComponent? pain = null)
+    {
+        if (!Resolve(uid, ref pain))
+            return;
+
+        pain.PainModificators.Remove(mod);
+        UpdateCurrentPainPercentage(pain);
+        Dirty(uid, pain);
+    }
+    private void UpdateCurrentPainPercentage(PainComponent comp)
+    {
+        var maxPainReductionModificatorStrength = FixedPoint2.Zero;
+        var painIncrease = FixedPoint2.Zero;
+        var painIncreases = comp.PainModificators.Where(mod => mod.Type == PainModificatorType.PainIncrease);
+        var painReductions = comp.PainModificators.Where(mod => mod.Type == PainModificatorType.PainReduction);
+        // get max pain reduction, sum pain increase
+        if (painIncreases.Any())
+            painIncrease = painIncreases.Select(mod => mod.EffectStrength).Sum();
+        if (painReductions.Any())
+            maxPainReductionModificatorStrength = painReductions.Max(mod => mod.EffectStrength);
+
+        var realCurrentPain = comp.CurrentPain + painIncrease;
+        // Pain reduction effectiveness linear decreases as the pain goes up
+        var newPainReduction = FixedPoint2.Max(0, -realCurrentPain * comp.PainReductionDecreaceRate + maxPainReductionModificatorStrength);
+        comp.CurrentPainPercentage = FixedPoint2.Clamp(realCurrentPain - newPainReduction, 0, 100);
+    }
+
+    private void UpdateCurrentPain(PainComponent comp, Dictionary<string, FixedPoint2> damageDict)
+    {
+        var newCurrentPain = FixedPoint2.Zero;
+        foreach (var (type, _) in damageDict)
+        {
+            if (_bruteTypes.Contains(type))
+            {
+                newCurrentPain += comp.BrutePainMultiplier * damageDict[type];
+            }
+
+            if (_burnTypes.Contains(type))
+            {
+                newCurrentPain += comp.BurnPainMultiplier * damageDict[type];
+            }
+
+            if (_toxinTypes.Contains(type))
+            {
+                newCurrentPain += comp.ToxinPainMultiplier * damageDict[type];
+            }
+
+            if (_airlossTypes.Contains(type))
+            {
+                newCurrentPain += comp.AirlossPainMultiplier * damageDict[type];
+            }
+        }
+
+        comp.CurrentPain = newCurrentPain;
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<PainComponent>();
+        while (query.MoveNext(out var uid, out var pain))
+        {
+            if (_mobState.IsDead(uid))
+                continue;
```

Comment:
This check should be done after the time check, also pull _timing.CurTime out to a time variable that is calculated before the loop so that you can reuse it